homeassistant/components/zha/fan.py

```diff
@@ -33,9 +33,13 @@
     SPEED_SMART
 ]
 
-VALUE_TO_SPEED = {i: speed for i, speed in enumerate(SPEED_LIST)}
-SPEED_TO_VALUE = {speed: i for i, speed in enumerate(SPEED_LIST)}
-
+SEQUENCE_LIST = [
+    [ SPEED_OFF, SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH, SPEED_SMART ],
+    [ SPEED_OFF, SPEED_LOW, SPEED_HIGH, SPEED_SMART ],
+    [ SPEED_OFF, SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH, SPEED_AUTO, SPEED_SMART ],
+    [ SPEED_OFF, SPEED_LOW, SPEED_HIGH, SPEED_AUTO, SPEED_SMART ],
```

Comment:
Only speed constants in the base fan component is allowed to be returned in the speed list or accepted as set speed argument.

Reply:
How do we go about when platform has more options than allowed by HA component? Like in this case, Zigbee "fan mode" has two more modes/speeds, values `5` and `6` 
![image](https://user-images.githubusercontent.com/5826160/61678777-d993df80-acd1-11e9-892e-7836c249e510.png)

and in the same time, climate component has a few more `fan` states (BTW what does `FAN_MIDDLE` means in climate?)

Reply:
Either leave some modes out, or have a non 1:1 mapping, ie map eg two device speeds to the same home assistant speed and pick one of the two device speeds to map to the home assistant speed the other way.

There's no really good solution for this problem currently. We have to solve it as best possible while staying in line with current architecture.